### ISSUE TITLE
ER: Extend Data Guard Association to support creation on VM DB Systems

### ISSUE THREAD

**Issue by simonhaslam** [oracle/oci-ansible-modules#106]
## Enhancement Request

According to https://oracle-cloud-infrastructure-ansible-modules.readthedocs.io/en/latest/modules/oci_data_guard_association_module.html#oci-data-guard-association-module for the creation_type "ExistingDbSystem is the only supported value".

For DBCS DB Systems running on VM, not bare metal, the only way you can create a DG standby is when the OCI API creates the standby DB System plus standby database in one operation (https://docs.cloud.oracle.com/en-us/iaas/Content/Database/Tasks/usingdataguard.htm "When you enable Oracle Data Guard for a virtual machine DB system database, a new DB system with the standby database is created and associated with the primary database."). This means that, unless I'm missing something, it's not possible to use the OCI Ansible module to set up Data Guard on VM DB Systems.

**Expected behavior**
It would be highly desirable to be able to use the ansible module to create a standby database for a VM-based DB System. Ideally creation type would support New, along with the other parameters needed. Most of those parameters are already in the API except perhaps availability domain and subnet-id OCID (e.g. when I look at the OCI CLI "oci db data-guard-association create with-new-db-system"). Plus, with the new OCI feature to have cross-region DG, maybe there will need to be a region parameter too (or maybe the AD covers that sufficiently).

**Environment**

* OS version:

* Ansible version:
 2.9.6

* OCI Python SDK version:
2.11.0

* OCI Ansible Modules version:
Name: oci
Version: 2.12.2
Summary: Oracle Cloud Infrastructure Python SDK

**Comment by nalsaber** [comment]
@simonhaslam Thanks for reporting this. Ansible module data-guard-association is missing the option to create a NewDbSystem. We will work on adding it in the coming releases.
https://docs.cloud.oracle.com/en-us/iaas/api/#/en/database/20160918/datatypes/CreateDataGuardAssociationWithNewDbSystemDetails

**Comment by veriton** [comment]
A colleague on my team at eProseed has enhanced the code to support DG Association on DBCS VM DB Systems (I saw Nabeel's comment on his PR, thanks), including adding parameters for the peer shape and something else we needed. So far we've tested successfully on single instance VM2.1 across 2 ADs in a single region, and the same for RAC is currently provisioning so seems likely to work too. What we haven't considered or tested is cross-region DG setup (new OCI feature from a few weeks ago) but I'm not sure that's in the Ansible module yet.
